### ISSUE TITLE
Ranged weapons shouldn't have a damage stat

### ISSUE THREAD

**Issue by honestaly** [TCArknight/T2K4E#13]
In truth, Damage, Crit, and Penetration come from the ammunition.  I noticed a damage stat on the ammo, but the others aren't there... not sure how that might complicate things, but I started doing prelimnary work to convert the T2K 1e and 2e weapons to 4e and damage is 100% based on the ammunition used it seems.

**Comment by honestaly** [comment]
This also solve the problem of weapons that use multiple ammo types (mortars, tank cannons, howitzers, etc) They can have multiple ammos types.  We just remove the damage, crit, and armor stats off of the ranged weapons and move it to ammo.

**Comment by TCArknight** [comment]
Yep. :) Both the weapon and the ammo would have the same fields available, and the weapon would use its default value or whatever ammo is in it. 

Also, just checked in where I’m at on the vehicle stuff. We just need to create Attribute things for each of the stats as
We’ll as the fields on the VehicleBase component for them.

**Comment by honestaly** [comment]
Before I start going crazy... is that how it actually works now?  Or is this an addition we need ot make?  :)

**Comment by honestaly** [comment]
I went crazy... I found the code to apply the damage from the ammo to a gun without a damage stat and that made things easy.

**Comment by honestaly** [comment]
Ok I thought the scripts that were there for damage would actually do what they seemed to do.  But After creating some civilian weapons that used Shotgun ammo, and updating the damage/crit/armor values for 12GA the values don't seem to pull to the character.

**Comment by honestaly** [comment]
Ok I thought the scripts that were there for damage would actually do what they seemed to do. But After creating some civilian weapons that used Shotgun ammo, and updating the damage/crit/armor values for 12GA the values don't seem to pull to the character.

**Comment by honestaly** [comment]
@TCArknight The more I think about this... it seems the right way to go.  But I see a couple problems.
I don't think the character builder page supports it.  I'm trying to think in the future... at some point FL will remember that .50 BMG can also be .50 SLAP and will likely put out that second ammo for the gun.  If they do how would that dispaly on the character editor as we have it now?  How would we want it to display?  For that matter, at some point they'll realize that Shotguns have both a Slug and a Shot round (not to mention the differences between 10GA, 12GA and 16GA) which would add a bunch of mildly different damage profiles I think?

I mean Ideally, the gun would show the damage profile for the ammo currently loaded into it.  But I don't think right now I'm pretty sure the code as written doesn't support that.  Do you have any further thoughts?

**Comment by TCArknight** [comment]
I'm taking a look at the Weapon display. :) 

If the weapon entry were 2 lines, what would you rather see on each line?

**Comment by TCArknight** [comment]
Take a look at the M4A1 in the Core_Weapons file I just pushed. 

For a Ranged weapon to contain ammo:
1 - Add the Ammo gizmo to the weapon. 
2 - Add the thing.holder tag to the weapon
3 - Add the appropriate tags to the HoldLimit entry (for ex: AmmoCat.Firearm & AmmoCatType.Rifle & AmmoType.amT2K55645 for the M4A1).

Then, for scripting, if the gizmo is holding any ammo, can pull the values from the ammo into the weapon.

**Comment by honestaly** [comment]
> I'm taking a look at the Weapon display. :)
> 
> If the weapon entry were 2 lines, what would you rather see on each line?

I think if a weapon was able to hold multiple ammos it would display the presently loaded ammo.
The ammo should probably show it's own stats for damage, armor pen, crit and blast

**Comment by honestaly** [comment]
> Take a look at the M4A1 in the Core_Weapons file I just pushed.
> 
> For a Ranged weapon to contain ammo:
> 1 - Add the Ammo gizmo to the weapon.
> 2 - Add the thing.holder tag to the weapon
> 3 - Add the appropriate tags to the HoldLimit entry (for ex: AmmoCat.Firearm & AmmoCatType.Rifle & AmmoType.amT2K55645 for the M4A1).
> 
> Then, for scripting, if the gizmo is holding any ammo, can pull the values from the ammo into the weapon.

So I see the gizmo and the like in the editor/vscode.  The damage though was coming from the weapon, not the ammo.
![image](https://user-images.githubusercontent.com/88744339/130110616-d473273e-dc72-4e18-aa43-b46fb4132167.png)

With the Gizmo enabled on the file... should't the little tool/gizmo button show up for the gun and allow selection of ammo?

**Comment by honestaly** [comment]
Also the M4A1 shows the following problems... instead of using the ByAmmo helper you added it just displays this.

![image](https://user-images.githubusercontent.com/88744339/132616243-00d47b9a-2e50-4f49-83df-767855cc7b72.png)

**Comment by TCArknight** [comment]
pushed fix for this. :)

**Comment by honestaly** [comment]
Sort of... this is working (I think) for single ammo weapons.  I.e. the M9 Pistol and M4A1 don't show any damage though even though.  It does something really weird with Multi-ammo weapons though.. like the 60mm Mortar

![image](https://user-images.githubusercontent.com/88744339/132707808-cea03a22-3669-4902-b582-6bf9cc9e94d8.png)

**Comment by TCArknight** [comment]
For the M4a1, have you clicked to store the ammo in the M4A1? :) 

Not sure on that mortar, maybe we do need multiple lines for each weapon. Is that the ammo as part of the name?

**Comment by honestaly** [comment]
Ok... storing the ammo works for the M4A1.  This doesn't mean in the combat tracker that this works though... I can't store 10,000 rounds of 5.56x45mm ammo into an M4A1 it only takes 30 at a time.  But that's way more complicated than where we are.

The mortars are goofy.  What I did was make a bunch of 60mm mortar ammo, and then check each of them as a possible ammo for the mortar.

![image](https://user-images.githubusercontent.com/88744339/132715620-d64da5ce-dfa3-4348-975a-f37993b3419b.png)

When adding ammo to a character, it displays funny for me.. if I select the ammo and add it, everything is fine.. the issue is only when adding it to a character.  It never displays this way on the character... just in that gear browser.

![image](https://user-images.githubusercontent.com/88744339/132715849-3a58c757-4133-4298-a54f-743d4677b480.png)

**Comment by honestaly** [comment]
However, what you did with the M4A1 works (more or less) so I'll work on copying it into every gun.  I had hoped we could make this work with the generic gun object instead, so that they all just behaved this way... Rather than having to modify all of the weapons to support this.

**Comment by honestaly** [comment]
So the current fix really works well... all it needs is the tags that make it the holder and DamageByAmmo, which I modified the weapon definition to assign.

The Mortar really hates it when you load ammo into it though... because of all the possible ammos.  I've only loaded HE into this one... even though I have 1 round of each type.

I'm trying to figure out how to access the assigned ammo types and then update the hold limit to just those.  Any ideas?

![image](https://user-images.githubusercontent.com/88744339/132862989-7ed9f196-fb58-42ba-9611-53e5c8cab60e.png)
